Packages/com.llealloo.audiolink/Runtime/Scripts/AudioLinkMidiControl.cs

```diff
@@ -0,0 +1,337 @@
+﻿
+using UdonSharp;
+using UnityEngine;
+using VRC.SDKBase;
+using VRC.Udon;
+
+namespace AudioLink
+{
+
+    [UdonBehaviourSyncMode(BehaviourSyncMode.None)]
+    public class AudioLinkMidiControl : UdonSharpBehaviour
+    {
+        public AudioLinkController _AudioLinkController;
+        public bool allowMidi;
+        public int midiChannel;
+
+        public bool _debugMidi;
+
+        void Start()
+        {
+            
+        }
+
+        public override void MidiNoteOn(int channel, int key, int value) {
+
+            
+
+            // Check On Midi
+            if (allowMidi && channel == midiChannel && Networking.IsOwner(_AudioLinkController.gameObject) && VRC.SDKBase.Utilities.IsValid(_AudioLinkController)) {
+
+                if (VRC.SDKBase.Utilities.IsValid(_AudioLinkController.themeColorController)) {
+
+                    ThemeColorController _ThemeColorController = _AudioLinkController.themeColorController;
+                
+                    switch (key) {
+
+                        case (int)MidiIds.ColorChord:
+                        // Enable Color Chord
+                        _ThemeColorController.themeColorDropdown.value = 0;
+                        break;
+
+                        case (int)MidiIds.Reset:
+                        // Trigger Controller Reset
+                        _AudioLinkController.ResetSettings();
+                        _ThemeColorController.ResetThemeColors();
+                        break;
+
+                        default:
+                        // No-op
+                        break;
+
+                    }
+
+                }
+
+            }
+
+            if (_debugMidi) {
+
+                //Debug.Log(channel);
+                Debug.Log(key);
+                Debug.Log(value);
+
+            }
+
+        }
+
+        public override void MidiNoteOff(int channel, int key, int value) {
+
+            // Check Off Midi
+            if (allowMidi && channel == midiChannel && Networking.IsOwner(_AudioLinkController.gameObject) && VRC.SDKBase.Utilities.IsValid(_AudioLinkController)) {
+
+                if (VRC.SDKBase.Utilities.IsValid(_AudioLinkController.themeColorController)) {
+
+                    ThemeColorController _ThemeColorController = _AudioLinkController.themeColorController;
+                    
+                    switch (key) {
+
+                        case (int)MidiIds.ColorChord:
+                        // Disable Color Chord
+                        _ThemeColorController.themeColorDropdown.value = 1;
```

Comment:
This will only work with the old controller. Note how ThemeColorController.cs does not define themeColorDropdown anywhere, because it isn't a dropdown anymore